packages/spor-datepicker-react/src/DatePicker.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useRef } from "react";
+import { AriaDatePickerProps } from "react-aria";
+import { Calendar } from "./Calendar";
+import { CalendarTriggerButton } from "./CalendarTriggerButton";
+import { DateField, StyledField } from "./DateField";
+
+type DatePickerProps = AriaDatePickerProps<DateValue> & {
+  variant: ResponsiveValue<"simple" | "with-trigger">;
+};
+/**
+ * A date picker component.
+ *
+ * There are two versions of this component – a simple one, and one with a trigger button for showing the calendar. Use whatever fits your design.
+ *
+ * ```tsx
+ * <DatePicker label="Dato" variant="simple" />
+ * ```
+ */
+export function DatePicker({ variant, ...props }: DatePickerProps) {
+  const state = useDatePickerState({
+    ...props,
+    shouldCloseOnSelect: true,
+  });
+  const ref = useRef(null);
+  const {
+    groupProps,
+    labelProps,
+    fieldProps,
+    buttonProps,
+    dialogProps,
+    calendarProps,
+  } = useDatePicker(props, state, ref);
+
+  const responsiveVariant =
+    useBreakpointValue(typeof variant === "string" ? [variant] : variant) ??
+    "simple";
+
+  const styles = useMultiStyleConfig("Datepicker", {
+    variant: responsiveVariant,
+  });
+
+  return (
+    <Box position="relative" display="inline-flex" flexDirection="column">
+      <Popover
+        {...dialogProps}
+        isOpen={state.isOpen}
+        onClose={() => state.setOpen(false)}
+        onOpen={() => state.setOpen(true)}
+        closeOnBlur
+        closeOnEsc
+        returnFocusOnClose
+      >
+        <InputGroup
+          {...groupProps}
+          ref={ref}
+          width="auto"
+          display="inline-flex"
+        >
+          <PopoverAnchor>
+            <StyledField
+              variant={responsiveVariant}
+              onClick={() => {
+                if (responsiveVariant === "simple") {
+                  state.setOpen(true);
+                }
+              }}
+            >
+              {responsiveVariant === "simple" && (
+                <CalendarOutline24Icon mr={2} alignSelf="center" />
+              )}
+              <Box>
+                <FormLabel {...labelProps} sx={styles.inputLabel}>
+                  {props.label}
+                </FormLabel>
+                <Box
+                  onKeyPress={(e) => {
+                    if (
+                      e.key === "Enter" &&
+                      !state.isOpen &&
+                      responsiveVariant === "simple"
+                    ) {
+                      // Don't submit the form
+                      e.stopPropagation();
+                      state.setOpen(true);
+                    }
+                  }}
+                >
+                  <DateField {...fieldProps} />
+                </Box>
+              </Box>
+            </StyledField>
+          </PopoverAnchor>
+          {responsiveVariant === "with-trigger" && (
+            <CalendarTriggerButton {...buttonProps} />
+          )}
```

Comment:
Er det mulig å dele opp denne litt, f.eks skille `simple` fra `with-Trigger`, slik at det blir lettere å lese?

Reply:
Ah, dele opp DatePicker og DateRangePicker  i simple og og with-trigger-komponenter? Det går an, men det blir litt mer duplisering av kode. Jeg skal ta en titt på disse filene for å forenkle dem litt uansett jeg tror jeg – det er en smule rotete nå.

Reply:
Jeg gjorde et forsøk, men det ble egentlig vel så forvirrende når de skal deles på tvers av DatePicker og DateRangePicker. Forenkla litt hvordan det ser ut nå, men tror det er så bra jeg vil gjøre det nå.

Reply:
Det hjalp litt ja 😄